unlock-app/src/components/content/certification/CertificationDetails.tsx

```diff
@@ -1,3 +1,359 @@
+            professional network.
+          </span>
+        )
+      } else if (account === key?.owner) {
+        return (
+          <span>
+            Here is the certificate you have received. This image is as
+            off-chain image to display in a certificate format. Please refer the
+            NFT link for on-chain validation.
+          </span>
+        )
+      } else {
+        return (
+          <span>
+            You are viewing the certificate issued to the recipient.{' '}
+            <Link
+              className="font-semibold text-gray-800 hover:text-brand-ui-primary"
+              href="/certification"
+            >
+              Learn more
+            </Link>{' '}
+            about Certification by Unlock Labs.
+          </span>
+        )
+      }
+    } else if (key) {
+      if (isLockManager) {
+        return (
+          <span>
+            Here is the template when recipient receive certificate from you.
+            They can connect wallet at this page & share the certification image
+            to the professional network.
+          </span>
+        )
+      } else {
+        return (
+          <span>
+            You are viewing a off-chain certification sample. Please connect
+            your wallet if you have received the certificate NFT or{' '}
+            <Link
+              className="font-semibold text-gray-800 hover:text-brand-ui-primary"
+              href="/certification"
+            >
+              Learn more
+            </Link>{' '}
+            about Certification by Unlock Labs.
+          </span>
+        )
+      }
+    } else {
+      return <span>No valid certification</span>
+    }
+  }
+
+  const canShareOrDownload =
+    (key?.owner === account || isLockManager) && key && !isPlaceholderData
+
+  const showCertification = key || (tokenId && key)
+
+  const injectedProvider = selectProvider(config)
+  const paywallConfig = {
+    locks: {
+      [lockAddress]: {
+        network,
+        emailRequired: true,
+      },
+    },
+  }
+
+  const badge =
+    isLockManager || !account
+      ? isLockManager
+        ? 'Template'
+        : 'Sample'
+      : undefined
+
+  const expiration =
+    key?.expiration !== MAX_UINT
+      ? isPlaceholderData
+        ? key?.expiration
+        : expirationAsDate(key?.expiration)
+      : undefined
+
+  return (
+    <main className="mt-8">
+      <Modal isOpen={isCheckoutOpen} setIsOpen={setCheckoutOpen} empty={true}>
+        <Checkout
+          injectedProvider={injectedProvider as any}
+          paywallConfig={paywallConfig}
+          handleClose={() => setCheckoutOpen(false)}
+        />
+      </Modal>
+
+      <div className="flex flex-col gap-6">
+        <Header />
+        {showCertification && (
+          <Certificate
+            tokenId={key?.tokenId}
+            network={network}
+            name={certificationData.name}
+            description={
+              // @ts-expect-error
```

Comment:
Please add the error for clarity!
```suggestion
              // @ts-expect-error ....ERROR....
```